platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -67,6 +84,9 @@ def __init__(self):
         # Initialize Platform name
         self.platform_name = device_info.get_platform()
 
+        # Initialize DMI data
```

Comment:
Do we have such init on different APIs? i think eeprom or maybe transceiver has it as well.
Is it ok to have the same init on different APIs? if so, no problem. lets just make sure it is.
Also, recently I saw other PRs to improve the init as some APIs can be from pmon and some from the host, so need to see if such improvement is relevant here as well. Maybe not

Reply:
Do you mind linking in the PRs that you have seen that may be relevant? At least using the method that I am there will be no problem performing this on multiple APIs as far as linux is concerned it is reading a file.

Reply:
I added lazy loading, I believe this should cover this concern. The DMI will now only be read when the hardware revision is specifically requested AND the dmi has not already been read.